main/simulator.cpp

```diff
@@ -1 +1,99 @@
 #include "main/simulator.h"
+
+std::vector<std::unique_ptr<objects::EnergyCollector>>
+buildCollectors(const AbstractModel &model, int numCollectors) {
+
+  // We need to check if model is empty, because we can't construct energy
+  // collectors around the model that doesn't exist.
+
+  if (model.empty()) {
+    throw std::invalid_argument("Empty model");
+  }
+
+  // Obviously num of collectors less then 4 cannot be allowed
+  if (numCollectors < 4) {
+    std::stringstream ss;
+    ss << "Num collectors: " << numCollectors << " is less them 1";
+    throw std::invalid_argument(ss.str());
+  }
+
+  // There are two possible configurations of EnergyCollectors:
+
+  // a) when numCollectors is not even, we have one collector at Vec3(0, 0,
+  // height), where height is equal to 4 * max(modelHeight, modelSide) and rest
+  // is equally distributed on 4 sides of the model, in a way, that distance
+  // between Vec3(0, 0, 0) and each collector remains always equal. Thats why
+  // we need to make sure that for this case numCollectors - 1 % 4 == 0.
+
+  // b) second configuration is similar to the first one, but there is no
+  // Energy collector at center (at Vec3(0, 0, height)). So to make sure that
+  // this configuration is possible, we need to check if numCollectors % 4 == 0.
+
+  // This leaves two possible cases:
+  // 1. numCollectors % 4 == 0
+  // 2. numCollectors - 1 % 4 == 0
+
+  if (!(numCollectors % 4 == 0 || (numCollectors - 1) % 4 == 0)) {
+    std::stringstream ss;
+    ss << "invalid numCollectors value \n"
```

Comment:
Change to:

```
"numCollectors or numCollectors-1 has to be divisible by 4, got numCollectors = " << numCollectors.
```

I assume that you know what % is a remainder of division, and == 0 means "divisible", right?